preview/MsixCore/msixmgr/Package.cpp

```diff
@@ -148,30 +155,115 @@ HRESULT MsixCoreLib::PackageBase::ProcessPSFIfNecessary()
         TraceLoggingWideString(m_relativeExecutableFilePath.c_str(), "PSF Executable"),
         TraceLoggingWideString(GetPackageDirectoryPath().c_str(), "path"));
 
-    // By default the PSF information is in config.json, but could be in a different json file.
-    for (auto& p : std::experimental::filesystem::directory_iterator(GetPackageDirectoryPath()))
+    // PSF config information should be in a file called config.json, so look for that file
+    std::wstring jsonConfigFile = GetPackageDirectoryPath() + psfConfigFile;
+    bool configFileExists = false;
+    RETURN_IF_FAILED(FileExists(jsonConfigFile, configFileExists));
+
+    if (configFileExists)
     {
-        if (std::experimental::filesystem::is_regular_file(p.path()) && CaseInsensitiveEquals(p.path().extension(), L".json"))
+        // parse the file and see if it has info we need.
+        std::ifstream ifs(jsonConfigFile);
+        IStreamWrapper isw(ifs);
+
+        Document doc;
+        doc.ParseStream(isw);
+
+        if (doc.HasParseError())
         {
-            // parse the file and see if it has info we need.
-            // TODO: fill this in with actual stuff. Hardcoding for this test package for now
-            std::wstring psfExecutable = L"Nokia Siemens Networks\\Managers\\BTS Site\\BTS Site Manager\\jre\\1_6_0\\bin\\javaw.exe";
-            std::wstring psfWorkingDirectory = L"Nokia Siemens Networks\\Managers\\BTS Site\\BTS Site Manager";
-            m_executionInfo.commandLineArguments = L"-splash:Splash_Wn_BTS_Site_Manager.png -cp cl\\cl.jar -client com.nokia.em.poseidon.PoseidonStarter -confFile cl\\CLConf.xml";
-
-            // resolve the given paths from json into full paths.
-            m_executionInfo.resolvedExecutableFilePath = FilePathMappings::GetInstance().GetExecutablePath(psfExecutable, m_packageFullName.c_str());
-            m_executionInfo.workingDirectory = FilePathMappings::GetInstance().GetExecutablePath(psfWorkingDirectory, m_packageFullName.c_str());
-
-            TraceLoggingWrite(g_MsixTraceLoggingProvider,
-                "PSF redirection",
-                TraceLoggingWideString(m_executionInfo.resolvedExecutableFilePath.c_str(), "Resolved PSF executable"),
-                TraceLoggingWideString(m_executionInfo.workingDirectory.c_str(), "WorkingDirectory"),
-                TraceLoggingWideString(m_executionInfo.commandLineArguments.c_str(), "Arguments"));
+            return HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED);
         }
-    }
+        if (doc.HasMember("applications"))
+        {
+            const Value& apps = doc["applications"];
+
+            for (Value::ConstValueIterator itr = apps.Begin(); itr != apps.End(); ++itr)
+            {
+                std::string id = (*itr)["id"].GetString();
+                std::wstring jsonApplicationId;

```

Comment:
>jsonApplicationId [](start = 29, length = 17)

add a TraceLogging here to show what applications we parsed from json

Reply:
and then another one after the for loop finishes to say that we went through all the applications and couldn't find the ID.

---
In reply to: [340917766](https://github.com/microsoft/msix-packaging/pull/249#discussion_r340917766) [](ancestors = 340917766)

Reply:
Added tracelogging for each app id found and for when none is found as well.